lib/vquic/quiche.c

```diff
@@ -199,6 +203,24 @@ CURLcode Curl_quic_connect(struct connectdata *conn, curl_socket_t sockfd,
     return CURLE_OUT_OF_MEMORY;
   }
 
+/* TODO: detect UNIX nicely */
+#ifndef _WIN32
+  qlog_dir = getenv("QLOGDIR");
+  if(qlog_dir) {
+    qlog_filename = malloc(strlen(qlog_dir) + 1 + sizeof(qs->scid) + 5);
+    /* TODO: better string concatentation and actually use the hex-formatted SCID */
+    p = qlog_filename;
+    strcpy(p, qlog_dir);
+    strcat(p, "/");
+    strcat(p, "12345678");
+    strcat(p, ".qlog");
```

Comment:
This could be made into a single `aprintf()` call that returns the alloced new string.